snapcraft/linters/linters.py

```diff
@@ -157,17 +157,21 @@ def _ignore_matching_filenames(
     issues: List[LinterIssue], *, lint: Optional[projects.Lint]
 ) -> None:
     """Mark any remaining filename match as ignored."""
-    if lint and lint.ignore.files:
-        for issue in issues:
-            for pattern in lint.ignore.files:
-                if (
-                    issue.filename
-                    and issue.result != LinterResult.IGNORED
-                    and fnmatch.fnmatch(issue.filename, pattern)
-                ):
-                    emit.verbose(
-                        f"Ignore {issue.name!r} linter issue ({issue.filename!r} "
-                        f"matches {pattern!r})"
-                    )
-                    issue.result = LinterResult.IGNORED
-                    break
+    if lint is None:
```

Comment:
with the removal of files is it possible for lint to be None now?

Reply:
yes: It's a bit confusing but `lint` here refers to the `lint:` key in the project, so it's still optional.
The function that calls this is called by lifecycle like this:
```python
    if command_name in ("pack", "snap"):
        issues = linters.run_linters(lifecycle.prime_dir, lint=project.lint)
        status = linters.report(issues, intermediate=True)